### DISCUSSION TITLE
CommandHandler returning timeout when hitting external API

### DISCUSSION THREAD

**Discussion by shapiroj18** [python-telegram-bot/python-telegram-bot#2375] [Q&A]
Hey folks, I'm having an issue where I am trying to hit an api in a CommandHandler and getting the context.error returning "timed out." Any ideas here on how to extend the timeout? The api is working as expected and returns 200 on other platforms.

**Comment by zeroone2numeral2** [comment]
If by "hitting an api" you mean calling a `Bot` method (or using an `Update` shortcut, such as `update.message.reply_text()`), then every method accepts an additional parameter `timeout`, which as far as I remember it's 20 seconds by default. You could try to increase it and see if it helps. What methods are you trying to use, that results in a `telegram.error.TimedOut` exception? 20 seconds should be enough for most operations

**Reply by shapiroj18** [reply]
Thanks for the response - the function I am trying to run can be found here - https://pastebin.com/sVpAMcbD.

I am calling an external api using httpx. For example, trying to get the data back form https://phishjams-dev.herokuapp.com/randomjam.

**Reply by zeroone2numeral2** [reply]
If the timeout is raised by the call to the external API, then it's kinda offtopic here. You have to ask to the maintainers of that API or try to raise the timeout threshold for the library you're using to send the request (see https://www.python-httpx.org/advanced/#timeout-configuration)

**Reply by shapiroj18** [reply]
hmmm yeah it does seem to be mainly a local issue with ngrok, so I think we can disregard. thanks for your help!
